# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.cpp
# Commit: fc9ffe1ae7ee
# Full Hash: fc9ffe1ae7ee592931f6adfb3d490ada4dbce43d
# Author: stransky <stransky@redhat.com>
# Date: 2024-07-08 16:33:59
# Regressor Bug: 1898476
# File Overlap Count: 1
# Description:
#   Bug 1898476 [Wayland] Provide surface lock by GtkCompositorWidget r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D214884
# ==============================================================================

diff -r 3e6bd6c75492 -r fc9ffe1ae7ee widget/gtk/GtkCompositorWidget.cpp
--- a/widget/gtk/GtkCompositorWidget.cpp	Mon Jul 08 08:01:08 2024 +0000
+++ b/widget/gtk/GtkCompositorWidget.cpp	Mon Jul 08 08:01:08 2024 +0000
@@ -211,5 +211,11 @@
 }
 #endif
 
+#if defined(MOZ_WAYLAND)
+UniquePtr<MozContainerSurfaceLock> GtkCompositorWidget::LockSurface() {
+  return mWidget->LockSurface();
+}
+#endif
+
 }  // namespace widget
 }  // namespace mozilla